sympy/simplify/tests/test_simplify.py

```diff
@@ -84,6 +85,38 @@ def test_ratsimpmodprime():
         y/2
 
 
+def test_rootofsimp():
+    r = RootOf(x**5 - 5*x + 12, 0)
+    s = RootOf(x**4 - x**2 + 1, 0)
+
+    f, g = r**5, 5*r - 12
+
+    assert f != g and rootofsimp(f) == g
+
+    f, g = r**10, 25*r**2 - 120*r + 144
+
+    assert f != g and rootofsimp(f) == g
+
+    f, g = r**4*(1 + r), r**4 + 5*r - 12
+
+    assert f != g and rootofsimp(f) == g
+
+    # test modular inversion
+    f, g = 1/r, -r**4/12 + 5/12
+
+    assert f != g and rootofsimp(f) == g
+
+    f, g = -r**4/12 + 5/12, r
+
+    assert f*g != 1 and rootofsimp(f*g) == 1
+
+    # test multiple roots in expression
```

Comment:
Good idea. A quick test I just performed shows that everything works as expected: that is, simplification is performed per root even if they're from the same poly. Thankfully, `expr.find(RootOf)` distinguishes roots based on their index as well.